cmd/utils.go

```diff
@@ -67,28 +67,47 @@ func request2BucketObjectName(r *http.Request) (bucketName, objectName string) {
 	if err != nil {
 		logger.CriticalIf(context.Background(), err)
 	}
-	return urlPath2BucketObjectName(path)
-}
-
-// Convert url path into bucket and object name.
-func urlPath2BucketObjectName(path string) (bucketName, objectName string) {
-	if path == "" || path == SlashSeparator {
+	return path2BucketObject("", path)
+}
+
+var path2BucketObjectPool = sync.Pool{
+	New: func() interface{} {
+		sbuf := make([]string, 2)
+		return &sbuf
+	},
+}
+
+// Similar to https://golang.org/pkg/strings/#SplitN
+// but doesn't trucate the return list, keeps the
+// size of the array 'n'
+func path2BucketObject(b, s string) (bucket, prefix string) {
+	sbuf := path2BucketObjectPool.Get().(*[]string)
```

Comment:
A am not sure why this is needed at all. It does nothing that 2 `string` values cannot do. 

Without this, it would be 32 bytes that could be allocated on the stack, but now it is forced to the heap and requires synchronization and a defer call.

Remove `path2BucketObjectPool` completely.

Reply:
@klauspost this is an allocation that I am avoiding. This call is called inside a recursive function across files and folders in the walk routine invoking allocations. There is a different way which this can be solved, let me add that.

Reply:
Fixed